### DISCUSSION TITLE
Feature Request: Use expressions for `database-path` in ATTACH

### DISCUSSION THREAD

**Discussion by aspyct** [duckdb/duckdb#13509] [Ideas]
I have a scenario where I want to attach to multiple postgres databases using the CLI. However, it seems impossible to do so without exposing the password into the `.duckdb_history` file.

There are two main supported ways to attach to a postgres database:

```
attach 'host=... user=... password=... dbname=...' as postgres_1 (type postgres);
```

and

```
-- Set the environment variables for postgres connection
attach '' as postgres_1 (type postgres);
```

While the second option works perfectly fine when you need to connect to one database, it doesn't work when you want to connect to multiple databases at the same time. You're left with only the first option, which will expose the password into your history file (to my knowledge, there is no way to disable history, but that would be a separate ticket).

I tried using expressions in `attach` to use custom environment variables, but it doesn't seem to be accepted.

```
attach 'host=localhost dbname=postgres user=postgres password=' || getenv('POSTGRES_PASSWORD') as postgres_2 (type postgres);
Parser Error: syntax error at or near "||"
```

Is there any way I can specify the password safely? If so, how, and if not, maybe that would be a good feature to have :)

**Comment by tmontes** [comment]
Joining in to say I'd find this useful, too!

**Comment by Tishj** [comment]
Does SET VARIABLE not work here?
I imagine you can construct your connection string and save that in a variable, then use `getvariable('my_connection_string)` in ATTACH

But perhaps ATTACH doesnt support constant parameters like that yet

**Reply by tmontes** [reply]
It apparently does not:
```
~ $ duckdb
v1.1.2 f680b7d08f
Enter ".help" for usage hints.
Connected to a transient in-memory database.
Use ".open FILENAME" to reopen on a persistent database.
D
D SET VARIABLE conn_str='';
D ATTACH getvariable('conn_str') AS psql (TYPE postgres);
Parser Error: syntax error at or near "getvariable"
LINE 1: ATTACH getvariable('conn_str') AS psql (TYPE p...
```

**Reply by tmontes** [reply]
PS: Slightly off-topic, the same could be said about `CREATE SECRET`: being able to use `getenv` in that context, or `getvariable` for that matter!

**Comment by malcook** [comment]
FWIW: I have a different use case than protecting secrets for wanting this feature.

In its most simple form, I'd like to write a linux command-line duckdb script to copy selected tables from one database to another destination database which I want to specify on the command line.

Without such a feature, it appears my only option is to write out the SQL statements needed to effect the attach, and then .read them back in.
